_overviews/toolkit/munit-assertions.md

```diff
@@ -1,18 +1,218 @@
+assert(obtainedResults.forall(x => x % 2 == 0))
+```
+{% endtab %}
+{% endtabs %}
+
+In case of failure, the test report looks like the following:
+
+```text
+# MyTests:
+# ==> X MyTests.some-unit-test  0.035s munit.FailException: ./example/MyTests.test.scala:9 assertion failed
+# 8:     val obtainedResults = input.map(double)
+# 9:     assert(obtainedResults.forall(x => x % 2 == 0))
+# 10:  }
+#     at munit.FunSuite.assert(FunSuite.scala:11)
+#     at MyTests.$init$$$anonfun$1(my-tests.test.scala:9)
+```
+
+## `assert` vs `assertEquals`
+
+You may be tempted to write `assert(a == b)` instead of `assertEquals(a, b)`.
+What is the difference between both?
+
+The difference lies in the way failures are reported. With `assertEquals`,
+MUnit shows a “diff” describing the difference between the expression that
+was obtained and the expression that was expected. On the other hand, with
+`assert`, MUnit only tells you that “an assertion failed”.
+
+Below is a concrete example, for the sake of comparison. First, with `assertEquals`:
+
+```text
+# MyTests:
+# ==> X MyTests.some-unit-test  0.068s munit.ComparisonFailException: ./example/MyTests.test.scala:7
+# 6:  test("some-unit-test") {
+# 7:    assertEquals(double(21), 42)
+# 8:  }
+# values are not the same
+# => Obtained
+# 23
+# => Diff (- obtained, + expected)
+# -23
+# +42
+```
+
+Then, the same example with `assert` shows less details about the failure:
+
+```text
+# MyTests:
+# ==> X MyTests.some-unit-test  0.068s munit.FailException: ./example/MyTests.test.scala:7 assertion failed
+# 6:  test("some-unit-test") {
+# 7:    assertEquals(double(21) == 42)
+# 8:  }
+```
+
+Note that it is possible to slightly improve the errors reported by assertions
+by adding [clues]({% link _overviews/toolkit/munit-clues.md %}).
+
+In case the obtained and expected results are complex objects, MUnit shows a detailed
+“diff” highlighting the very fields that were different between them. Here is an example
+of test specification involving complex objects:
+
+{% tabs assertions-3 %}
+{% tab 'Scala 2 and 3' %}
+```scala
+// File User.scala
+case class User(name: String, email: String)
+```
+```scala
+// Test specification
+val obtained = User("Alice", "alice@gmail.com")
+val expected = User("Alice", "alice@sca.la")
+assertEquals(obtained, expected)
+```
+{% endtab %}
+{% endtabs %}
+
+The test report is the following:
+
+```text
+# MyTests:
+# ==> X MyTests.some-unit-test  0.072s munit.ComparisonFailException: /tmp/tests/my-tests.test.scala:11
+# 10:    val expected = User("Alice", "alice@sca.la")
+# 11:    assertEquals(obtained, expected)
+# 12:  }
+# values are not the same
+# => Obtained
+# User(
+#   name = "Alice",
+#   email = "alice@gmail.com"
+# )
+# => Diff (- obtained, + expected)
+#    name = "Alice",
+# -  email = "alice@gmail.com"
+# +  email = "alice@sca.la"
+#  )
+```
+
+You see that the test reports points out that the field `email` you obtain is
+different from your expectation.
```

Comment:
IMO these examples are not really necessary. People will see by themselves that the error reports are good.

Reply:
Removed in 23521c6